src/service/routing/directions.clj

```diff
@@ -65,21 +70,25 @@
    Example:
    (direction graph :coordinates [{:lon 1 :lat 2} {:lon 3 :lat 4}]"
   [graph & parameters]
-  (let [{:keys [coordinates steps]} parameters
-        start     (brute-nearest graph (first coordinates))
-        dst       (brute-nearest graph (last coordinates))
-        traversal (alg/dijkstra graph :value-by length
-                                      :start-from #{(key start)})
-        trace     (reduce (fn [_ trace] (when (= (key trace) (key dst)) (reduced trace)))
-                          nil
-                          traversal)]
-    (if (nil? trace)
-      {:code "NoRoute"}
-      {:code "Ok"
-       :waypoints (map (fn [point] {:name "wonderland"
-                                    :location [(:lon point) (:lat point)]})
-                       coordinates)
-       :routes [(route graph trace)]})))
+  (if-not (validate-radiuses parameters)
```

Comment:
@mehdisadeghi sorry my mistake. I meant rather to move the `routes` to the `core` ns. That way you can have a spec only namespace which doesnt rely on any other ns, thus avoiding the cyclic dependency